### PR TITLE
fixes bug 1183838 - Upgrade to Django 1.7

### PR DESCRIPTION
**Pull request by peterbe**
@AdrianGaudebert r?

So this upgrade trigger a lot of package updates. 
I created a brand new virtualenv based on **python 2.7** (which is a must) and one of the problems is that django's runserver command tries to import `inotify`. When it does so, because it is in the virtualenv, it causes a `SystemExit` and kills runserver. I'm going to update @rhelmer's fork of inotify. 

One of the reasons the PR looks big and scary is because I had to do `git mv crashstats/auth crashstats/authentication` because you can't have two installed apps that have the same suffix. We have `django.contrib.auth` AND `crashstats.auth`.

### CODE REVIEW

**Comment by peterbe** [comment]
One thing that might need to be done is change the sha with which we point to @rhelmer's pyinotify fork once this lands https://github.com/rhelmer/pyinotify/pull/2

**Comment by peterbe** [comment]
I realize that I haven't tested the wsgi endpoint. I've only tested it with runserver.

**Comment by peterbe** [comment]
Note-to-self: Here's how I started the wsgi app locally:
```
uwsgi --pythonpath ./webapp-django/ -w webapp-django.wsgi.socorro-crashstats --http :9090 -H ~/virtualenvs/socorro
```
Then visiting `localhost:9090`

**Comment by peterbe** [comment]
@AdrianGaudebert if it wasn't clear. This guy's ready for review again.

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ cache:
     - $HOME/.cache/pip
 language: python
 python:
-  - "2.6"
+  - "2.7"
```
Comment (AdrianGaudebert) [inline .travis.yml]: Are we there yet? Is it official? :pray:

File: webapp-django/crashstats/settings/base.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
     'django.contrib.sessions',
     'django.contrib.staticfiles',
     'commonware.response.cookies',
-    'south',  # important that django_nose comes AFTER this one
+    # 'south',  # important that django_nose comes AFTER this one
```
Comment (AdrianGaudebert) [inline webapp-django/crashstats/settings/base.py]: Why not just remove it since it's in django now?

File: .travis.yml (hunk 1 above)
Comment (rhelmer) [inline .travis.yml]: @AdrianGaudebert Yep, we've been 2.7 only since the AWS switch.

Comment (peterbe) [inline .travis.yml]: @rhelmer I just noticed that 
```
docs/development/build-pkg-from-source.rst
docs/development/codeanddatabaseupdate.rst
docs/development/dev-services.rst
```
Also speak of Python 2.6. 
Can I just blanket update them too?

File: webapp-django/crashstats/settings/base.py (hunk 2 above)
Comment (peterbe) [inline webapp-django/crashstats/settings/base.py]: A perfectly good review comment. 
I started that way but clearly didn't get back to removing it properly. 
Thanks!

File: .travis.yml (hunk 1 above)
Comment (rhelmer) [inline .travis.yml]: @peterbe please do.

**Comment by AdrianGaudebert** [comment]
@peterbe so what's up with ``pyinotify``? I don't see it being updated in the requirements.

**Comment by peterbe** [comment]
> @peterbe so what's up with pyinotify? I don't see it being updated in the requirements. 

Another great piece of PR review right there! I made a patch to Rob's fork but I was waiting to update `requirements.txt` until my [PR](https://github.com/rhelmer/pyinotify/pull/2) got merged. 
Thanks!

File: .travis.yml (hunk 1 above)
Comment (AdrianGaudebert) [inline .travis.yml]: :sunny:

**Comment by AdrianGaudebert** [comment]
Tested locally, works well! r+

**Pull request merged**